src/evict/evict_file.c

```diff
@@ -62,9 +62,13 @@ __wt_evict_file(WT_SESSION_IMPL *session, WT_CACHE_OP syncop)
          * sets of files open, updates in a no-longer-referenced file may not yet be visible, and
          * the write will fail with EBUSY. Our caller handles that error, retrying later.
          */
-        if (syncop == WT_SYNC_CLOSE && __wt_page_is_modified(page))
-            WT_ERR(__wt_reconcile(session, ref, NULL,
-              WT_REC_EVICT | WT_REC_HS | WT_REC_CLEAN_AFTER_REC | WT_REC_VISIBLE_ALL));
+        if (syncop == WT_SYNC_CLOSE && __wt_page_is_modified(page)) {
+            rec_flags = WT_REC_EVICT | WT_REC_CLEAN_AFTER_REC | WT_REC_VISIBLE_ALL;
+            if (!WT_IS_HS(btree) && !WT_IS_METADATA(dhandle))
+                rec_flags |= WT_REC_HS;
+
```

Comment:
Two minor things: 1. I think this deserves a comment. 2. Please remove the blank line.